### PR TITLE
Deprecate TiDB Binlog

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

This PR adds deprecation warnings or notes to docs related to TiDB Binlog

Note: The status of TiDB Binlog in [TiDB Features | PingCAP Docs](https://docs.pingcap.com/tidb/dev/basic-features) will be updated in another PR that is particularly for updating the status of all v8.3 features on that page. So it is not updated in this PR to avoid conflict.

image

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18346
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-flashback-table.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ FlashbackToNewName ::=
 
 If a table is dropped and the GC lifetime has passed, you can no longer use the `FLASHBACK TABLE` statement to recover the dropped data. Otherwise, an error like `Can't find dropped / truncated table 't' in GC safe point 2020-03-16 16:34:52 +0800 CST` will be returned.
 
-Pay attention to the following conditions and requirements when you enable TiDB Binlog and use the `FLASHBACK TABLE` statement:
+Pay attention to the following conditions and requirements when you enable TiDB Binlog (deprecated starting from v8.3.0) and use the `FLASHBACK TABLE` statement:
```
Comment (benmeadowcroft) [inline sql-statements/sql-statement-flashback-table.md:46]: I would suggest just using "deprecated"

File: tidb-binlog-deployment-topology.md [hunk 2]
```diff
@@ -6,9 +6,11 @@ aliases: ['/docs/dev/tidb-binlog-deployment-topology/']
 
 # TiDB Binlog Deployment Topology
 
-This document describes the deployment topology of [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md) based on the minimal TiDB topology.
+This document describes the deployment topology of [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md) based on the minimal TiDB topology. TiDB Binlog provides near real-time backup and replication.
 
-TiDB Binlog is the widely used component for replicating incremental data. It provides near real-time backup and replication.
+> **Warning:**
+>
+> Starting from TiDB v8.3.0, TiDB Binlog is deprecated, and is planned to be removed in a future release. Use [TiCDC](/ticdc/ticdc-overview.md) instead for incremental data replication.
```
Comment (benmeadowcroft) [inline tidb-binlog-deployment-topology.md:13]: Replace:
> TiDB Binlog is deprecated

With:
> TiDB Binlog is fully deprecated

File: tidb-binlog/bidirectional-replication-between-tidb-clusters.md [hunk 3]
```diff
@@ -9,9 +9,8 @@ aliases: ['/docs/dev/tidb-binlog/bidirectional-replication-between-tidb-clusters
 > **Warning:**
 >
 > - Currently, bidirectional replication is still an experimental feature. It is **NOT** recommended to use it in the production environment.
+> - Starting from TiDB v8.3.0, [TiDB Binlog](/tidb-binlog/tidb-binlog-overview.md) is deprecated, and is planned to be removed in a future release. Use [TiCDC](/ticdc/ticdc-overview.md) instead for incremental data replication. Use [Point-in-Time Recovery (PITR)](/br/br-pitr-guide.md) for the PITR scenario needs.
```
Comment (benmeadowcroft) [inline tidb-binlog/bidirectional-replication-between-tidb-clusters.md:12]: Replication was deprecated in v7.5.0, not v8.3.0, so this specific content about replication should reflect that

File: tidb-binlog/bidirectional-replication-between-tidb-clusters.md [hunk 4]
```diff
@@ -9,9 +9,8 @@ aliases: ['/docs/dev/tidb-binlog/bidirectional-replication-between-tidb-clusters
 > **Warning:**
 >
 > - Currently, bidirectional replication is still an experimental feature. It is **NOT** recommended to use it in the production environment.
```
Comment (benmeadowcroft) [inline tidb-binlog/bidirectional-replication-between-tidb-clusters.md:11]: This specific line should be removed due to the improvements in FD-1689. I am not sure if this is tracked in a different PR?

File: faq/deploy-and-maintain-faq.md [hunk 5]
```diff
@@ -113,6 +113,6 @@ The Direct mode wraps the Write request into the I/O command and sends this comm
 
 ## What public cloud vendors are currently supported by TiDB?
 
-TiDB supports deployment on [Google Cloud GKE](https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-on-gcp-gke), [AWS EKS](https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-on-aws-eks), and [Alibaba Cloud ACK](https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-on-alibaba-cloud).
+TiDB supports deployment on [Google Cloud GKE](https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-on-gcp-gke), [AWS EKS](https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-on-aws-eks), and [Alibaba Cloud ACK](https://docs.pingcap.com/tidb-in-kubernetes/v1.5/deploy-on-alibaba-cloud).
```
[the remark this answers is not in the document]
Comment (lilin90) [reply faq/deploy-and-maintain-faq.md:116]: Just the original link is broken in the auto check. Later we added an aliases for the corresponding doc. No need to update this link in this PR now. I'll remove the change.

File: tidb-binlog/bidirectional-replication-between-tidb-clusters.md (hunk 4 above)
Comment (lilin90) [reply tidb-binlog/bidirectional-replication-between-tidb-clusters.md:11]: It seems that the line was ignored in https://github.com/pingcap/docs/pull/18568. I'll follow up.

Comment (lilin90) [reply tidb-binlog/bidirectional-replication-between-tidb-clusters.md]: This line was removed in https://github.com/pingcap/docs/pull/18606.

**Comment by lilin90** [comment]
Thanks @benmeadowcroft! All your comments have been addressed.

**Review by benmeadowcroft — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

**Pull request merged**